lib/smart_proxy_pulp_plugin/pulp3_client.rb

```diff
@@ -0,0 +1,43 @@
+require 'net/http'
+require 'net/https'
+require 'uri'
+require 'smart_proxy_pulp_plugin/settings'
+
+module PulpProxy
+  class Pulp3Client
+    def self.get(path)
+      uri = URI.parse(::PulpProxy::Pulp3Plugin.settings.pulp_url.to_s)
+      req = Net::HTTP::Get.new(URI.join(uri.to_s.chomp('/') + '/', path))
+      req.add_field('Accept', 'application/json')
+      req.content_type = 'application/json'
+      self.http.request(req)
+    end
+
+    def self.capabilities
+      body = JSON.parse(get("api/v3/status/").body)
+      body['versions'].map{|item| item['component'] }
+    rescue => e
+      []
+    end
+
+    def self.http
```

Comment:
This code is duplicated in many places. I wonder if the smart proxy should provide a helper for this

Reply:
I know its duplicated in the pulp2 and pulp3 code, is it duplicated somewhere else?  i intentionally duplicated a lot of the code since the pulp2 code will be likely removed at some point in the future.

Reply:
I think that other smart proxy modules and plugins have very similar code.

Reply:
it looks like the main smart-proxy has its central version of this code, but its intent is to talk to the foreman server.  Looking at this method i think some of the options here (ca, ssl client certs) really don't apply to talking to the pulp server and should be removed.  I might could see an argument for the ca file, but i think relying on the system ca would be better than the foreman server's ca.   The client cert piece doesn't make any sense at all, so i will remove that.  

In the end i think this method will be very small here.   Thoughts?

Reply:
It's probably worth looking into in the future. Even a parametrized function that to connect to some URL with certs would be useful.

Reply:
:+1:  i just simplified the function for now